react-native-community/cli
# refactor: move pods installation from `init` to ios-specific commands

Summary:
---------
Currently Pods are installed during `init` command, what makes project setup much longer than it supposed to be. This change makes pods installation optional during the `init` command by displaying the prompt, and moves the installation process to the `run-ios` and `build-ios` commands. Pods are automatically installed when running ios-specific commands and:
- they were not installed before
- a new dependency/devDependency was introduced in `package.json`
- new flag `--force-pods` was used, which forces to install pods (might be helpful when there are any errors related to pods)

Additionally, new flag `--install-pods` was introduced to `init` command, to avoid displaying prompt. If set to `true`, pods will install, `false` will skip the installation, and when not used, prompt will be displayed.

To install pods during `*-ios` command, it compares md5 hashes of native dependencies names and its versions. To achieve this, I added one new field to dependency config: `version` which holds the version of the library from package.json/podspec file.

## Demo
**1. `init` with prompt**

https://github.com/react-native-community/cli/assets/13985840/1028821d-8a6d-4361-be85-c37f6eca5bbc

**2. `init` with `--install-pods false` - skip prompt and pods installation**

https://github.com/react-native-community/cli/assets/13985840/cde70ccd-a310-42b8-8d54-fa18fc1d36d4


**3. `run-ios` for the first time** (cut the video because it takes too long 😄, app runs after installation)

https://github.com/react-native-community/cli/assets/13985840/a50fabb7-7092-402e-8cb2-3f4b32a9ec6c




Test Plan:
----------



1. Follow the contributing instructions
2. Init new project - it should display prompt about pods installation
3. Press `y` to confirm - pods should be installed
4. Init new project once again - it should display prompt
5. Press 'n' - pods installation should be skipped
6. Init new project with `--install-pods` flag - pods should be installed automatically
7. Init new project with `--install-pods false` - prompt & pods installation should be skipped
8. Run `run-ios` command and verify pods are installed before building the app
9. Run `run-ios` again and verify app is built without installing pods
10. Install any random dependency, e.g. `@react-navigation/native`
11. Run `run-ios` again and verify pods are installed before building the app
12. Run `run-ios` again and verify app is built without installing pods
13. Run `run-ios --force-pods` and verify pods are installed before building the app

Checklist
----------
